File: 이것이 코딩테스트다/03.DFS&BFS/5.BFS.py
from collections import deque

graph = [
  [],
  [2, 3, 8],
  [1, 7],
  [1, 4, 5],
  [3, 5],
  [3, 4],
  [7],
  [2, 6, 8],
  [1, 7],
]

visited = [False] * 9
q = deque()

def bfs (start):
  visited[start] = True
  q.append(start)

  while q:
    v = q.popleft()
    print("방문", v)
    for node in graph[v]:
      if not visited[node]:
        q.append(node)
        visited[node] = True
bfs(1)

File: 이것이 코딩테스트다/03.DFS&BFS/5.BFS.js
const graph = [
  [],
  [2, 3, 8],
  [1, 7],
  [1, 4, 5],
  [3, 5],
  [3, 4],
  [7],
  [2, 6, 8],
  [1, 7],
];
let visited = Array.from({ length: 9 }, () => false);
let q = [];
const bfs = (start) => {
  q.push(start);
  visited[start] = true;

  while (q.length) {
    const v = q.shift();
    console.log("방문", v);

    for (const node of graph[v]) {
      if (!visited[node]) {
        visited[node] = true;
        q.push(node);
      }
    }
  }
};
bfs(1);
